nevergrad/optimization/test_recaster.py

```diff
@@ -81,3 +81,22 @@ def test_recast_optimizer_and_stop() -> None:
     optimizer = FakeOptimizer(instrumentation=2, budget=100)
     optimizer.ask()
     # thread is not finished... but should not hang!
+
+def test_provide_recommendation() -> None:
+    nevergrad_optimizer = optimizerlib.OnePlusOne(instrumentation=2, budget=100)
+    # even if no ask&tell, recommendation should work:
+    assert nevergrad_optimizer.provide_recommendation() is not None
+
+    # the recommended solution should be the better one among
+    # all told solutions for OnePlusOne solver:
+    x1 = nevergrad_optimizer.ask()
+    nevergrad_optimizer.tell(x1, 10)
+    x2 = nevergrad_optimizer.ask()
+    nevergrad_optimizer.tell(x2, 5)
+    recommended_solution = nevergrad_optimizer.provide_recommendation()
+    for i in range(len(recommended_solution.data)):
+        assert recommended_solution.data[i] == x2.data[i]
```

Comment:
Please directly compare as arrays (and not element against element with the for loop), this is faster to write, faster to read, and faster to execute.
Also, `data`  is bound to disappear with the new parametrization system (see #323 #391), so the best way to replace this currently would be:
```python
np.testing.assert_array_equal(recommended_solution.args[0], x2.args[0])
```
After the change of parametrization, we will be able to write `value` instead of `args[0]`, that should be simpler :)